.github/workflows/push-image.yml

```diff
@@ -4,54 +4,107 @@ on:
-      push_to_dockerhub: ${{ steps.parse_configs.outputs.push_to_dockerhub}}
+      DOCKERHUB_ORG: ${{ steps.get-dockerhub-namespace.outputs.DOCKERHUB_ORG }}
+      push_to_gcr: ${{ steps.parse_configs.outputs.push_to_gcr }}
+      push_to_dockerhub: ${{ steps.parse_configs.outputs.push_to_dockerhub }}
+      tag: ${{ steps.event.outputs.tag }}
+      repo_name: ${{ steps.registry-repo.outputs.repo_name }}
+      stacks: ${{ steps.get-stacks.outputs.stacks }}
     steps:
     - name: Checkout
       uses: actions/checkout@v4
 
-    - name: Set matrix
-      id: set-matrix
+    - name: Parse Event
+      id: event
       run: |
-        release_version="$(jq -r '.release.tag_name' "${GITHUB_EVENT_PATH}" | sed s/^v//)"
+        set -euo pipefail
+        shopt -s inherit_errexit
+        
+        # If the workflow has been triggered from dispatch event
+        if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+          echo "tag=${{ github.event.inputs.version }}" >> "$GITHUB_OUTPUT"
+        else #The workflow has been triggered from publish event
+          echo "tag=$(jq -r '.release.tag_name' "${GITHUB_EVENT_PATH}" | sed s/^v//)" >> "$GITHUB_OUTPUT"
+        fi
 
+    - name: Get Registry Repo Name
+      id: registry-repo
+      run: |
         # Strip off the org and slash from repo name
         # paketo-buildpacks/repo-name --> repo-name
-        repo_name=$(echo "${{ github.repository }}" | sed 's/^.*\///')
+        echo "repo_name=$(echo "${{ github.repository }}" | sed 's/^.*\///')" >> "$GITHUB_OUTPUT"
 
-        asset_prefix="${repo_name}-${release_version}-"
-        oci_images=$(jq -c --arg asset_prefix "$asset_prefix" '[.release.assets[].name | select(endswith(".oci")) | split(".oci") | .[0] | split($asset_prefix) | .[1]]' "${GITHUB_EVENT_PATH}")
-        printf "matrix=%s\n" "${oci_images}" >> "$GITHUB_OUTPUT"
+    - name: Get stacks 
+      id: get-stacks
+      run: |
+        stacks=$(
+          cat <<EOF
+        [
+          {
+            "name": "stack",
+            "config_dir": "stack",
+            "output_dir": "build",
+            "build_image": "build",
+            "run_image": "run",
+            "create_build_image": true
+          }
+        ]
+        EOF
+        )
+
+        if [[ -f ${{ env.STACKS_FILEPATH }} ]]; then
+          stacks=$(jq '[.images[] |
+          . +
+          {
+            "create_build_image": (.create_build_image // false)
+          }]' ${{ env.STACKS_FILEPATH }} )
+        fi
+
+        ## Due to publish.sh script compatibility, in case of there is no build image
+        ## set a default value to the default build image which will always be available.
+        stacks=$(echo "$stacks" | jq 'map(if .create_build_image == false then .build_image = "build" else . end)')
+
+        ## Filter stacks array to include the minimum number of attributes
+        stacks=$(echo "$stacks" | jq 'map({ name, build_image, run_image })')
```

Comment:
Is there any time where we expect there be stacks that don't have all of the required attributes?  Just asking to understand if maybe we should check/fail versus just filtering out?

Reply:
I think that's impossible

Reply:
I guess this is not filtering out stacks, just filtering down to the attributes we need?